widgets/BigQueryExecute-action.json

```diff
@@ -0,0 +1,139 @@
+    {
+      "label": "Basic",
+      "properties": [
+        {
+          "widget-type": "textbox",
+          "label": "Project ID",
+          "name": "project",
+          "widget-attributes": {
+            "default": "auto-detect"
+          }
+        },
+        {
+          "widget-type": "textarea",
+          "label": "SQL",
+          "name": "sql",
+          "widget-attributes": {
+            "placeholder": "Specifies the query to be executed."
+          }
+        }
+      ]
+    },
+    {
+      "label": "Advanced",
+      "properties": [
+        {
+          "name": "legacy",
+          "label": "Legacy",
+          "widget-type": "radio-group",
+          "widget-attributes": {
+            "layout": "inline",
+            "default": "false",
+            "options": [
+              {
+                "id": "true",
+                "label": "True"
+              },
+              {
+                "id": "false",
+                "label": "False"
+              }
+            ]
+          }
+        },
+        {
+          "name": "mode",
+          "label": "Mode",
+          "widget-type": "radio-group",
+          "widget-attributes": {
+            "layout": "inline",
+            "default": "batch",
+            "options": [
+              {
+                "id": "batch",
+                "label": "Batch"
+              },
+              {
+                "id": "interactive",
+                "label": "Interactive"
+              }
+            ]
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Dataset Name",
+          "name": "dataset",
+          "widget-attributes": {
+            "placeholder": "Name of dataset to store the results."
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Table Name",
+          "name": "table",
+          "widget-attributes": {
+            "placeholder": "Name of table to store the results."
+          }
+        },
+        {
+          "name": "cache",
+          "label": "Cache",
+          "widget-type": "radio-group",
+          "widget-attributes": {
+            "layout": "inline",
+            "default": "false",
+            "options": [
+              {
+                "id": "true",
+                "label": "True"
+              },
+              {
+                "id": "false",
+                "label": "False"
+              }
+            ]
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Job Location",
+          "name": "location",
+          "widget-attributes": {
+            "placeholder": "Location of the job.",
+            "default": "US"
+          }
+        },
+        {
+          "widget-type": "textbox",
+          "label": "Timeout",
+          "name": "timeout",
+          "widget-attributes": {
+            "placeholder": "Query timeout in minutes. Defaults to 10.",
+            "default": "10"
```

Comment:
I think we should pick defaults that are less likely to result in errors. 10 minutes seems really short. What is the default for the BigQuery client if no timeout is specified?
It may be better to allow this to be null, which means to never time out.

Reply:
Also, I think caching, legacy, and timeout would fit better under an 'Advanced' section, as they are all optional and require some understanding of BigQuery.